x/tax2gas/utils/fee_tax.go

```diff
@@ -0,0 +1,147 @@
+package utils
+
+import (
+	"regexp"
+	"strings"
+
+	errorsmod "cosmossdk.io/errors"
+	sdkmath "cosmossdk.io/math"
+	wasmtypes "github.com/CosmWasm/wasmd/x/wasm/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	authz "github.com/cosmos/cosmos-sdk/x/authz"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+
+	marketexported "github.com/classic-terra/core/v3/x/market/exported"
+	"github.com/classic-terra/core/v3/x/tax2gas/types"
+)
+
+var IBCRegexp = regexp.MustCompile("^ibc/[a-fA-F0-9]{64}$")
+
+func isIBCDenom(denom string) bool {
+	return IBCRegexp.MatchString(strings.ToLower(denom))
+}
+
+// FilterMsgAndComputeTax computes the stability tax on messages.
+func FilterMsgAndComputeTax(ctx sdk.Context, tk types.TreasuryKeeper, burnTaxRate sdk.Dec, msgs ...sdk.Msg) sdk.Coins {
```

Comment:
I am seeing two definitions of the function `FilterMsgAndComputeTax`. Please, remove the redundancy.